core/domain/story_services_test.py

```diff
@@ -277,825 +246,12 @@ def test_update_story_which_not_corresponding_topic_id(self):
-
-    def test_cannot_update_node_title_with_invalid_node_id(self):
-        change_list = [story_domain.StoryChange({
-            'cmd': story_domain.CMD_UPDATE_STORY_NODE_PROPERTY,
-            'property_name': story_domain.STORY_NODE_PROPERTY_TITLE,
-            'node_id': 'invalid_node',
-            'old_value': '',
-            'new_value': 'new_title'
-        })]
-
-        with self.assertRaisesRegexp(
-            Exception,
-            'The node with id invalid_node is not part of this story'):
-            story_services.update_story(
-                self.USER_ID, self.STORY_ID, change_list, 'Update node title.')
-
-    def test_cannot_delete_node_with_invalid_node_id(self):
-        change_list = [story_domain.StoryChange({
-            'cmd': story_domain.CMD_DELETE_STORY_NODE,
-            'node_id': 'invalid_node'
-        })]
-
-        with self.assertRaisesRegexp(
-            Exception,
-            'The node with id invalid_node is not part of this story'):
-            story_services.update_story(
-                self.USER_ID, self.STORY_ID, change_list, 'Delete node.')
-
-    def test_cannot_delete_starting_node_of_story(self):
-        changelist = [
-            story_domain.StoryChange({
-                'cmd': story_domain.CMD_ADD_STORY_NODE,
-                'node_id': self.NODE_ID_2,
-                'title': 'Title 2'
-            }),
-            story_domain.StoryChange({
-                'cmd': story_domain.CMD_UPDATE_STORY_NODE_PROPERTY,
-                'property_name': (
-                    story_domain.STORY_NODE_PROPERTY_DESTINATION_NODE_IDS),
-                'node_id': self.NODE_ID_2,
-                'old_value': [],
-                'new_value': [self.NODE_ID_1]
-            }),
-            story_domain.StoryChange({
-                'cmd': story_domain.CMD_UPDATE_STORY_NODE_OUTLINE_STATUS,
-                'node_id': self.NODE_ID_2,
-                'old_value': False,
-                'new_value': True
-            }),
-            story_domain.StoryChange({
-                'cmd': story_domain.CMD_UPDATE_STORY_CONTENTS_PROPERTY,
-                'property_name': (
-                    story_domain.INITIAL_NODE_ID),
-                'old_value': self.NODE_ID_1,
-                'new_value': self.NODE_ID_2
-            })
-        ]
-        story_services.update_story(
-            self.USER_ID, self.STORY_ID, changelist, 'Added node.')
-
-        change_list = [story_domain.StoryChange({
-            'cmd': story_domain.CMD_DELETE_STORY_NODE,
-            'node_id': self.NODE_ID_2
-        })]
-
-        with self.assertRaisesRegexp(
-            Exception,
-            'The node with id %s is the starting node for the story, '
-            'change the starting node before deleting it.' % self.NODE_ID_2):
-            story_services.update_story(
-                self.USER_ID, self.STORY_ID, change_list, 'Delete node.')
-
-    def test_delete_initial_node(self):
```

Comment:
If you have no story rights, make sure that rights for stories are still handled correctly, e.g. a story can only be edited by the topic manager.

Reply:
Yeah, that is done. The `check_can_edit_story` decorator checks whether the user can edit the corresponding topic.